Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift

```diff
@@ -2,47 +2,84 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
-import Foundation
+import Common
 import Redux
 
 protocol ThemeManagerProvider {
-    var systemThemeIsOn: Bool { get }
-    func setSystemTheme(isOn: Bool)
-    func setAutomaticBrightness(isOn: Bool)
-    func brightnessChanged(_ value: Float)
+    func getCurrentThemeManagerState() -> ThemeSettingsState
+    func toggleUseSystemAppearance(_ enabled: Bool)
+    func toggleAutomaticBrightness(_ enabled: Bool)
+    func updateManualTheme(_ theme: BuiltinThemeName)
+    func updateUserBrightness(_ value: Float)
 }
 
 class ThemeManagerMiddleware: ThemeManagerProvider {
-    var themeManager: LegacyThemeManager
+    var legacyThemeManager: LegacyThemeManager
+    var themeManager: ThemeManager
 
-    var systemThemeIsOn: Bool {
-        themeManager.systemThemeIsOn
-    }
-
-    init(themeManager: LegacyThemeManager = LegacyThemeManager.instance) {
-        // TODO: Add support for themeManager
+    init(legacyThemeManager: LegacyThemeManager = LegacyThemeManager.instance,
+         themeManager: ThemeManager = AppContainer.shared.resolve()) {
+        self.legacyThemeManager = legacyThemeManager
         self.themeManager = themeManager
     }
 
-    lazy var setSystemTheme: Middleware<AppState> = { state, action in
+    lazy var themeManagerProvider: Middleware<AppState> = { state, action in
         switch action {
-        case ThemeSettingsAction.enableSystemAppearance(let enabled):
-            self.themeManager.systemThemeIsOn = enabled
-            store.dispatch(ThemeSettingsAction.systemThemeChanged(enabled))
+        case ThemeSettingsAction.themeSettingsDidAppear:
+            let currentThemeState = self.getCurrentThemeManagerState()
+            DispatchQueue.main.async {
+                store.dispatch(ThemeSettingsAction.receivedThemeManagerValues(currentThemeState))
+            }
```

Comment:
Just thinking on this a little more, since we want it to be a convention that all messages passed back to the store should be on the main thread would it make sense to instead make the store a MainActor?
I'm not sure if that would cause other issues with this implementation tho. It could also be added as a future change, doesn't need to be in this PR, so maybe a spike to look into it would be helpful.

Reply:
Sounds like a good idea, I will create a SPIKE to investigate